core/src/main/java/org/ironjacamar/core/connectionmanager/pool/capacity/CapacityFactory.java

```diff
@@ -0,0 +1,239 @@
+
+import org.ironjacamar.common.api.metadata.common.Extension;
+import org.ironjacamar.core.CoreLogger;
+import org.ironjacamar.core.api.connectionmanager.pool.CapacityDecrementer;
+import org.ironjacamar.core.api.connectionmanager.pool.CapacityIncrementer;
+import org.ironjacamar.core.spi.classloading.ClassLoaderPlugin;
+import org.ironjacamar.core.util.Injection;
+
+import java.util.Map;
+
+import org.jboss.logging.Logger;
+
+/**
+ * The capacity factory
+ *
+ * @author <a href="mailto:jesper.pedersen@ironjacamar.org">Jesper Pedersen</a>
+ */
+public class CapacityFactory
+{
+   /** The logger */
+   private static CoreLogger log = Logger.getMessageLogger(CoreLogger.class, CapacityFactory.class.getName());
+
+   /**
+    * Constructor
+    */
+   private CapacityFactory()
+   {
+   }
+
+   /**
+    * Create a capacity instance based on the metadata
+    * @param metadata The metadata
+    * @param isCRI Is the pool CRI based
+    * @param classLoaderPlugin the ClassLoaderPlugin used to load incrementer/decrementer classes
+    * @return The instance
+    */
+   public static org.ironjacamar.core.connectionmanager.pool.Capacity create(
+         org.ironjacamar.common.api.metadata.common.Capacity metadata, boolean isCRI,
+         ClassLoaderPlugin classLoaderPlugin)
+   {
+      if (metadata == null)
+         return DefaultCapacity.INSTANCE;
+
+      CapacityIncrementer incrementer = null;
+      CapacityDecrementer decrementer = null;
+
+      // Incrementer
+      if (metadata.getIncrementer() != null && metadata.getIncrementer().getClassName() != null)
+      {
+         incrementer = loadIncrementer(metadata.getIncrementer(), classLoaderPlugin);
+
+         if (incrementer != null)
+         {
+            if (metadata.getIncrementer().getConfigPropertiesMap().size() > 0)
+            {
+               Injection injector = new Injection();
+
+               Map<String, String> properties = metadata.getIncrementer().getConfigPropertiesMap();
+               for (Map.Entry<String, String> property : properties.entrySet())
+               {
+                  try
+                  {
+                     injector.inject(incrementer, property.getKey(), property.getValue());
+                  }
+                  catch (Throwable t)
+                  {
+                     //TODO: log log.invalidCapacityOption(property.getKey(),
+                     //                          property.getValue(), incrementer.getClass().getName());
+                  }
+               }
+            }
+         }
+         else
+         {
+            //TODO: log log.invalidCapacityIncrementer(metadata.getIncrementer().getClassName());
+         }
+      }
+
+      if (incrementer == null)
+         incrementer = DefaultCapacity.DEFAULT_INCREMENTER;
+
+      // Decrementer
+      if (metadata.getDecrementer() != null && metadata.getDecrementer().getClassName() != null)
+      {
+         if (!isCRI)
```

Comment:
We shouldn't need the isCRI parameter any more, as the checks will be performed in the MCP impl